### PR TITLE
Adding xapi_bridge Configuration

### PR DESCRIPTION
**Pull request by thatguynishant**

### CODE REVIEW

**Comment by thatguynishant** [comment]
@bryanlandia @tkeemon I am letting the `templates/xapi_bridge.conf` stay in this PR. This file was config for supervisor. We can remove this later.

**Comment by thatguynishant** [comment]
[(8) ISC xAPI phase 3: Uptime monitoring of xAPI bridge](https://trello.com/c/lJu6AGJl/3516-8-isc-xapi-phase-3-uptime-monitoring-of-xapi-bridge)

File: playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=edX xAPI Bridge Service
+After=network.target
+Documentation=https://github.com/adlnet-archive/edx-xapi-bridge
```
Comment (bryanlandia) [inline playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2]: @thatguynishant Let's use our fork at https://github.com/appsembler/edx-xapi-bridge for the Documentation URL

File: playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+[Service]
+User=xapi
+WorkingDirectory=/edx/app/xapi/edx-xapi-bridge/
```
Comment (bryanlandia) [inline playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2]: Let's make use of the default vars to make this path dynamic if possible

File: playbooks/roles/xapi_bridge/tasks/main.yml [hunk 3]
```diff
@@ -60,3 +60,63 @@
...
+- name: create the log file
+  file:
+    path: "{{ COMMON_LOG_DIR }}/{{ xapi_bridge_user }}/{{ item }}"
+    state: touch
+    mode: 0644
+    owner: "{{ xapi_bridge_log_user }}"
+    group: "{{ xapi_bridge_log_user }}"
+  with_items:
+    - 'xapi_stderr.log'
+    - 'xapi_stdout.log'
```
Comment (bryanlandia) [inline playbooks/roles/xapi_bridge/tasks/main.yml]: In the python application, I have set logging to log everything to a single file, at level INFO 

```py
    if getattr(settings, 'DEBUG_MODE', False):
        logging.basicConfig(
            format='%(levelname)s:%(message)s',
            level=logging.DEBUG
        )
    else:
        logging.basicConfig(
        filename='/edx/var/log/xapi/xapi_bridge.log',
        filemode='a+',
        format='%(levelname)s:%(message)s',
        level=logging.INFO
)
```

This doesn't match your stderr + stdout capture setup here.  

What do you think we should do?  The non-DEBUG setup here will send all logged messages, info, warnings, handled errors to a single file.  It's possible to filter errors out to stderr I think but I think it's more work than necessary.

File: playbooks/roles/xapi_bridge/templates/xapi_bridge.conf [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[program:xapi_bridge]
```
Comment (bryanlandia) [inline playbooks/roles/xapi_bridge/templates/xapi_bridge.conf]: This .conf is for a Supervisor program still, right?  Shouldn't we get rid of this if we are going to use systemd?

File: playbooks/roles/xapi_bridge/defaults/main.yml [hunk 5]
```diff
@@ -5,10 +5,12 @@
 
 xapi_bridge_user: xapi
 xapi_bridge_user_group: adm  #perms on tracking.log
+xapi_bridge_log_user: root
```
Comment (bryanlandia) [inline playbooks/roles/xapi_bridge/defaults/main.yml:8]: I don't think the Python process will have permission to write to the log files if they are owned by root.

**Review by bryanlandia — CHANGES_REQUESTED** [review]
Hey @thatguynishant some questions for you here. Also, it would be great to rebase off appsember/ficus/master and squash the commits before we merge.

File: playbooks/roles/xapi_bridge/tasks/main.yml (hunk 3 above)
Comment (thatguynishant) [reply playbooks/roles/xapi_bridge/tasks/main.yml]: @bryanlandia yeah. I think what you're doing is correct. I wasn't aware of this. Let's wait for @tkeemon and see what he thinks about this.

File: playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2 (hunk 1 above)
Comment (thatguynishant) [reply playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2]: Updated.

File: playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2 (hunk 2 above)
Comment (thatguynishant) [reply playbooks/roles/xapi_bridge/templates/etc/systemd/system/xapi_bridge.service.j2]: Updated!

File: playbooks/roles/xapi_bridge/templates/xapi_bridge.conf (hunk 4 above)
Comment (thatguynishant) [reply playbooks/roles/xapi_bridge/templates/xapi_bridge.conf]: @bryanlandia yeah we're not using supervisor for managing the process. But I still wanted to keep this here to try a few things out. I'll remove this and keep it in a different branch.

File: playbooks/roles/xapi_bridge/defaults/main.yml (hunk 5 above)
Comment (bryanlandia) [reply playbooks/roles/xapi_bridge/defaults/main.yml:8]: OK, we talked about this and it's just for system-level logging.

File: playbooks/roles/xapi_bridge/tasks/main.yml (hunk 3 above)
Comment (bryanlandia) [reply playbooks/roles/xapi_bridge/tasks/main.yml]: Once again, system-level logging and application level logging are kept separate so it's fine what you're doing.

**Review by bryanlandia — APPROVED** [review]
looks good to me now

**Comment by thatguynishant** [comment]
[(3) ISC xAPI phase 3: Set up log rotation and log handling](https://trello.com/c/iRlDK96G/3517-3-isc-xapi-phase-3-set-up-log-rotation-and-log-handling)

**Comment by thatguynishant** [comment]
@bryanlandia I added the logrotate config. Would you be able to review it again?

**Review by bryanlandia — APPROVED** [review]
Looks good!

**Pull request merged**